### PR TITLE
Allow different async storage solutions

### LINKED ISSUE #6: Cannot use RN community AsyncStorage, allow for other solutions
Glad to see this package resurrected!  Thank you!

React Native developers using Expo (as of v37 - most current as of this issue creation) are stuck using the _depreciated_ React Native implementation of [AsyncStorage](https://docs.expo.io/versions/latest/react-native/asyncstorage).  There is no way (without ejecting) to use the [community](https://github.com/react-native-community/async-storage) version.

Allowing the developer to import their preferred AsyncStorage package and passing it via `options` in the `connect(endpoint, options)` method would allow Expo users to enjoy this package as well.  

Another package that has done this, is [Redux-Persist](https://github.com/rt2zz/redux-persist) and can provide a good example of implementation.

I will have a PR shortly to address this.  Or, if anyone has a better solution, I would be interested to hear it!  

Cheers.

### PR DESCRIPTION
**Pull request by PolarisWT**
Fixes #6
This update allows the developer to pass the AsyncStorage object to the `connect()` method with a fallback of the RN Community AsyncStorage if none is provided (backwards compatibility)

- [x] Updated Docs to show new usage.
- [x] Tries @react-native-community/async-storage as a fallback, moved to optional dependency
- [x] Bump version 2.1.0.  

Import AsyncStorage from:
- [@react-native-communiuty](https://github.com/react-native-community/async-storage)
    - Currently preferred library
- [react-native](https://docs.expo.io/versions/latest/react-native/asyncstorage/)
    - Expo (<= v37) uses RN AsyncStorage

Note: To run locally, I needed to:
- Add @babel/runtime
- `import Meteor, { Mongo, withTracker } from 'meteor-react-native'`
    - Note the unnamed `Meteor` import.  Got an error if it was { Meteor }...

### CODE REVIEW

File: src/user/User.js [hunk 1]
```diff
@@ -120,11 +118,14 @@ module.exports = {
...
+      value = await Data._options.AsyncStorage.getItem(TOKEN_KEY);
     } catch (error) {
       console.warn('AsyncStorage error: ' + error.message);
     } finally {
       this._loginWithToken(value);
     }
   },
+  setItem(){
+
+  }
```
Comment (TheRealNate) [inline src/user/User.js:130]: Hi! Quick question: has this method been added purposefully or is it old code?

Comment (PolarisWT) [reply src/user/User.js:130]: Missed that one.  Dead code, and I'll push a new update with it removed.  I was planning on abstracting the methods: `setItem`, `getItem`, and `removeItem` to allow even wider storage solutions, but thought better of it..

**Comment by TheRealNate** [comment]
Looks good! Merging your PR. The new version will be published on NPM within a few hours.

**Comment by TheRealNate** [comment]
The update `2.0.2` has been published on NPM

**Pull request merged**